R/UK.R

```diff
@@ -0,0 +1,519 @@
+#' United Kingdom Class for downloading, cleaning and processing notification
+#' data.
+#'
+#' @description Country specific information for downloading, cleaning
+#'  and processing covid-19 region data for an example Country.
+#'
+#' @details Inherits from `DataClass`
+#' @source https://coronavirus.data.gov.uk/details/download #nolint
+#' @examples
+#' \dontrun{
+#' region <- Italy$new(verbose = TRUE, steps = TRUE)
+#' region$download()
+#' region$clean()
+#' region$process()
+#' region$return()
+#' }
+Uk <- R6::R6Class("Uk", # rename to country name
+  inherit = DataClass,
+  public = list(
+    # Core Attributes (amend each paramater for country specific infomation)
+    #' @field level_1_region the level 1 region name.
+    level_1_region = "region", # add more levels as needed
+    #' @field level_2_region the level 2 region name
+    level_2_region = "authority",
+    #' @field data_url link to raw data
+    data_url = "https://api.coronavirus.data.gov.uk/v2/data",
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = list(
+      # Cases by date of specimen
+      "newCasesBySpecimenDate", "cumCasesBySpecimenDate",
+      # Cases by date of report
+      "newCasesByPublishDate", "cumCasesByPublishDate",
+      # deaths by date of report
+      "newDeaths28DaysByPublishDate", "cumDeaths28DaysByPublishDate",
+      # deaths by date of death
+      "newDeaths28DaysByDeathDate", "cumDeaths28DaysByDeathDate",
+      # Tests - all
+      "newTestsByPublishDate", "cumTestsByPublishDate",
+      # Hospital - admissions
+      "newAdmissions", "cumAdmissions",
+      #
+      # --- Additional non-standard variables --- #
+      # # # Hospital
+      # "cumAdmissionsByAge", "covidOccupiedMVBeds",
+      # "hospitalCases", "plannedCapacityByPublishDate",
+      # Tests by pillar
+      "newPillarOneTestsByPublishDate", "newPillarTwoTestsByPublishDate",
+      "newPillarThreeTestsByPublishDate", "newPillarFourTestsByPublishDate"
+    ),
+
+    #' @description Specific function for getting region codes for UK .
+    #' @rdname get_region_codes
+    get_region_codes = function() {
+      tar_level <- paste0("level_", self$level, "_region")
+      tar_level_name <- self[[tar_level]]
+      self$region <- list(country = self$country, level = tar_level_name)
+      self$region$code <- "ons_region_code"
+      self$region$codes_lookup <- NULL
+    },
+
+    #' @description UK specific download function
+    #' @importFrom purrr map
+    download = function() {
+      # set up filters
+      private$set_filters()
+      message("Downloading UK data.")
```

Comment:
I missed the issue with this but I assume there is a reason can't have a verbose trigger? 

If there is an issue does having a helper work? (

```
message_verbose <- function(..., verbose = FALSE) {
if (verbose) {
  message(...)
}
return(invisible(NULL))
}
```
}

Reply:
Due to the number of if statements I get the error from lintr about cyclomatic complexity, which seems to be about the number of paths through code. The limit is 15 and the extra ones for verbose push it past this.

Reply:
Regardless of if you nest in helper functions? Interesting. You should be able to change this in `.lintr`?

Reply:
Yeah happy to amend the .lintr, just thought I would check first. I thought about trying to redirect messages away from console using something like `sink` right at the start but then this redirects all messages.